dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -409,3 +439,279 @@ def __init__(self,
                                                          squeeze_labels=squeeze_labels,
                                                          dynamic_shape=dynamic_shape,
                                                          last_batch_padded = last_batch_padded)
+
+###############################################
+###############################################
+################## Gluon API ##################
+###############################################
+###############################################
+
+
+class SmartArray(object):
+    def __init__(self, array):
+        self._data = array.reshape(-1)
+        self._view = array
+
+    def resize(self, shape):
+        new_size = np.prod(shape)
+
+        if new_size > self._data.size:
+            self._data = mx.nd.zeros(new_size, self._data.context, dtype=self._data.dtype)
+            self._view = self._data
+        elif new_size < self._data.size:
+            self._view = self._data[:new_size]
+        else:
+            self._view = self._data
+        self._view = self._view.reshape(shape)
+        return self._view
+
+    @property
+    def array(self):
+        return self._view
+
+
+class DALIGluonGenericIterator(_DALIIteratorBase):
+    """
+    General DALI iterator for MXNet with Gluon API. It can return any number of
+    outputs from the DALI pipeline in the form of per GPU tuples of NDArray
+    of NDArrays or list of NDArrays (for output marked as DALIGluonGenericIterator.SPARSE_TAG).
+
+
+
+    Please keep in mind that NDArrays returned by the iterator are
+    still owned by DALI. They are valid till the next iterator call.
+    If the content needs to be preserved please copy it to another NDArray.
+
+    Parameters
+    ----------
+    pipelines : list of nvidia.dali.pipeline.Pipeline
+                List of pipelines to use
+    output_types : list of str
```

Comment:
Why str? Maybe some python enum?
Also why not add some 3rd value - like auto and iterator would return sparse when dense is not available?

Reply:
Tbh, I don't know, I just followed the existing iterators

Reply:
Sure. How about this 3rd option with auto?

Reply:
That's completely doable, I didn't want to make it implicit at first, but it can be something extra

Reply:
Just suggestion to consider, maybe surprising to the user and not useful.

Reply:
Actually, it might be good to have output_types=None by default where every element is a DENSE_TAG, so only users knowing what they do will use sparse batches

Reply:
Done